UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -246,6 +236,20 @@ public AgentInfo Info
 
         WriteAdapter m_WriteAdapter = new WriteAdapter();
 
+
+        /// Represents the reward the agent accumulated during the current step.
+        /// It is reset to 0 at the beginning of every step.
+        /// Should be set to a positive value when the agent performs a "good"
+        /// action that we wish to reinforce/reward, and set to a negative value
+        /// when the agent performs a "bad" action that we wish to punish/deter.
+        /// Additionally, the magnitude of the reward should not exceed 1.0
+        LowLevelRewardProviderComponent m_RewardProviderComponent;
+
+        public IRewardProvider rewardProvider
+        {
+            get { return m_RewardProviderComponent.GetRewardProvider(); }
```

Comment:
Should this be `m_RewardProviderComponent?.GetRewardProvider()` instead? The Debug.Asserts below mention "m_RewardProviderComponent is null..." but if that were the case, this would hit a null pointer dereference first.